File: algorithm/src/main/java/baekjoon/p01924/Main.java
package baekjoon.p01924;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt(), sc.nextInt()));
    }

    private static String solve(int x, int y) {
        int diff = 0;

        for (int i = 1; i < x; i++) {
            diff += day(i);
        }

        diff += y;
        return Week.get(diff % 7).toString();
    }

    private static int day(int month) {
        switch (month) {
            case 1:
            case 3:
            case 5:
            case 7:
            case 8:
            case 10:
            case 12:
                return 31;
            case 4:
            case 6:
            case 9:
            case 11:
                return 30;
            default:
                return 28;
        }
    }

    private enum Week {

        SUN, MON, TUE, WED, THU, FRI, SAT;

        static Week get(int week) {
            for (Week value : values()) {
                if (value.ordinal() == week) {
                    return value;
                }
            }

            return null;
        }
    }

}

File: algorithm/src/main/java/baekjoon/p01924/Main.kt
package baekjoon.p01924

fun main(args: Array<String>) {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(inputs[0], inputs[1]))
}

private fun solve(x: Int, y: Int): String {
    var diff = 0

    for (i in 1 until x) {
        diff += day(i)
    }

    diff += y
    diff %= 7

    for (value in Week.values()) {
        if (value.ordinal == diff) {
            return value.toString()
        }
    }

    return ""
}

private fun day(month: Int): Int {
    return when (month) {
        1, 3, 5, 7, 8, 10, 12 -> 31
        4, 6, 9, 11 -> 30
        else -> 28
    }
}

private enum class Week {

    SUN, MON, TUE, WED, THU, FRI, SAT;

}